src/main/java/org/opentripplanner/routing/algorithm/raptor/router/street/TransferToAccessEgressLegMapper.java

```diff
@@ -0,0 +1,33 @@
+package org.opentripplanner.routing.algorithm.raptor.router.street;
+
+import com.conveyal.r5.otp2.api.transit.TransferLeg;
+import org.opentripplanner.model.Stop;
+import org.opentripplanner.routing.algorithm.raptor.transit.Transfer;
+import org.opentripplanner.routing.algorithm.raptor.transit.TransitLayer;
+import org.opentripplanner.routing.algorithm.raptor.transit.request.TransferWithDuration;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class TransferToAccessEgressLegMapper {
+    private TransitLayer transitLayer;
+
+    public TransferToAccessEgressLegMapper(TransitLayer transitLayer) {
+        this.transitLayer = transitLayer;
+    }
+
+    public Collection<TransferLeg> map(Map<Stop, Transfer> input, double walkSpeed) {
+        List<TransferLeg> result = new ArrayList<>();
+        for (Map.Entry<Stop, Transfer> entry : input.entrySet()) {
+            Stop stop = entry.getKey();
+            Transfer transfer = entry.getValue();
+            int duration = (int)Math.floor(transfer.getDistanceMeters() / walkSpeed);
+            int stopIndex = transitLayer.getIndexByStop(stop);
+            Transfer newTransfer = new Transfer(stopIndex, transfer.getDistanceMeters(), transfer.getCoordinates());
+            result.add(new TransferWithDuration(newTransfer, duration));
```

Comment:
The [constructor](https://github.com/opentripplanner/OpenTripPlanner/pull/2754/files#diff-ffa2a1841de068a50122829a04a189a0R12) for `TransferWithDuration` expects `transfer, walkSpeed` as parameters.
```suggestion
            result.add(new TransferWithDuration(newTransfer, walkSpeed));
```